gluoncv/auto/estimators/image_classification/image_classification.py

```diff
@@ -483,9 +520,12 @@ def _predict(self, x, ctx_id=0, with_proba=False):
         x = x.as_in_context(self.ctx[ctx_id])
         pred = self.net(x)
         topK = min(5, self.num_class)
-        ind = nd.topk(pred, k=topK)[0].astype('int').asnumpy().flatten()
-        probs = mx.nd.softmax(pred)[0].asnumpy().flatten()
-        df = pd.DataFrame([{'class': self.classes[ind[i]], 'score': probs[ind[i]], 'id': ind[i]} for i in range(topK)])
+        if self._problem_type in [MULTICLASS, BINARY]:
+            ind = nd.topk(pred, k=topK)[0].astype('int').asnumpy().flatten()
```

Comment:
One problem here is that k can never >2 if it's BINARY classification.

Reply:
> One problem here is that k can never >2 if it's BINARY classification.

Should not be a problem as:
```
topK = min(5, self.num_class)
```
self.num_class = 2 for binary problems.